src/components/rest-client/hooks/useRestClient.ts

```diff
@@ -16,6 +21,7 @@ export const useRestClient = ({
   const [method, setMethod] = useState<THttpMethod>(paramsMethod);
   const [endpoint, setEndpoint] = useState(paramsUrl);
   const [headers, setHeaders] = useState<HeaderItemType[]>(queryHeaders);
+  const { variables } = useVariablesEditor();
```

Comment:
```suggestion
  const [variables, setVariables] = useState<VariableItemType[]>([]);

  useEffect(() => {
    const storageVariables = localStorage.getItem(LS_VARIABLES_KEY);
    if (storageVariables) {
      setVariables(JSON.parse(storageVariables));
    }
  }, []);
```
Сan add state only for variables so as not to pull the whole hook